src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -110,6 +112,9 @@ newtype StepProof (level :: *) (variable :: * -> *) =
     StepProof { getStepProof :: Seq (StepProofAtom level variable) }
   deriving (Eq, Show)
 
+instance Hashable.Hashable (StepProof level variable) where
+    hashWithSalt s _ = Hashable.hashWithSalt s (0 :: Int)
```

Comment:
Shouldn't this return the salt instead?
```suggestion
    hashWithSalt s _ = s
```

Reply:
This would result in a situation where two terms of different types have the same hash. Harmless, but I'm not sure it's better than giving it a different hash...